ExtractAWData/ExtractAWData.cs

```diff
@@ -0,0 +1,61 @@
+using Azure.Storage.Blobs;
+using Gwa.Etl.Models;
+using Gwa.Etl.Services;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Gwa.Etl
+{
+    public class ExtractAWData
+    {
+        private readonly BlobClient blobClient;
+        private readonly IConfiguration configuration;
+        private readonly IHttpClientFactory httpClientFactory;
+        private readonly ILogger<ExtractAWData> logger;
+
+        public ExtractAWData(BlobClient blobClient, IConfiguration configuration, IHttpClientFactory httpClientFactory, ILogger<ExtractAWData> logger)
+        {
+            this.blobClient = blobClient;
+            this.configuration = configuration;
+            this.httpClientFactory = httpClientFactory;
+            this.logger = logger;
+        }
+
+        [Function("ExtractAWData")]
+        public async Task Run([TimerTrigger("0 0 8 * * 0")] TimerInfo timerInfo)
+        {
+            try
+            {
+                logger.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}.");
+                logger.LogInformation($"Last execution was at {timerInfo.ScheduleStatus.Last}. Next execution will be at {timerInfo.ScheduleStatus.Next}");
+
+                AirWatchService getUsers = new(configuration, httpClientFactory, logger);
+                ProcessedUsers processedUsers = await getUsers.Process();
+
+                IDictionary<string, User> users = processedUsers.Users;
+                using (MemoryStream json = new(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(users.Values))))
+                {
+                    _ = await blobClient.UploadAsync(json, true);
```

Comment:
Is there a benefit to using the discard variable `_` over not assigning the return value to anything, i.e just `await blobClient.UploadAsync(json, true);`?

Reply:
I've added the discard as my IDE was prompting me to do so! I'm not sure if sonarcloud would also have complained. Even though I did it because it was the easy option I like the reason for doing it too i.e. it provides a clear intention to the reader (and the compiler) that the variable is unused.